app/controllers/devise_token_auth/registrations_controller.rb

```diff
@@ -28,68 +21,64 @@ def create
           status: 'error',
           data:   @resource.as_json,
           errors: [I18n.t("devise_token_auth.registrations.missing_confirm_success_url")]
-        }, status: 403
+        }, status: :bad_request
       end
 
       # if whitelist is set, validate redirect_url against whitelist
-      if DeviseTokenAuth.redirect_whitelist
-        unless DeviseTokenAuth.redirect_whitelist.include?(redirect_url)
-          return render json: {
-            status: 'error',
-            data:   @resource.as_json,
-            errors: [I18n.t("devise_token_auth.registrations.redirect_url_not_allowed", redirect_url: redirect_url)]
-          }, status: 403
-        end
+      if DeviseTokenAuth.redirect_whitelist &&
+         !DeviseTokenAuth.redirect_whitelist.include?(redirect_url)
+        return render json: {
+          status: 'error',
+          data:   @resource.as_json,
+          errors: [I18n.t("devise_token_auth.registrations.redirect_url_not_allowed", redirect_url: redirect_url)]
+        }, status: :forbidden
       end
 
-      begin
-        # override email confirmation, must be sent manually from ctrl
-        resource_class.skip_callback("create", :after, :send_on_create_confirmation_instructions)
-        if @resource.save
-          yield @resource if block_given?
-
-          unless @resource.confirmed?
-            # user will require email authentication
-            @resource.send_confirmation_instructions({
-              client_config: params[:config_name],
-              redirect_url: redirect_url
-            })
-
-          else
-            # email auth has been bypassed, authenticate user
-            @client_id = SecureRandom.urlsafe_base64(nil, false)
-            @token     = SecureRandom.urlsafe_base64(nil, false)
+      # override email confirmation, must be sent manually from ctrl
```

Comment:
Same as the above, based on the [ruby styleguide](https://github.com/bbatsov/ruby-style-guide#begin-implicit)